5_distance_analysis/nbconverted/1_centroid_distance.py

```diff
@@ -0,0 +1,681 @@
+    # convert dataframe to long-form to plot it
+    plot_df = (class_counts_gain_df[class_counts_gain_df.class_name == class_name]
+      .drop(columns='class_name')
+      .dropna(axis='index')
+      .reset_index()
+      .rename(columns={'index': 'identifier'})
+      .melt(id_vars='identifier', value_name='count', var_name='num_hits')
+    )
+    sns.kdeplot(data=plot_df, x='count', hue='num_hits', ax=ax)
+    ax.set_xlim(-10, 500)
+    ax.set_ylim(0.0, 0.03)
+
+
+# In[17]:
+
+
+sns.set({'figure.figsize': (18, 12)})
+
+fig, axarr = plt.subplots(2, 2)
+
+# plot copy loss results here
+for ix, class_name in enumerate(['class 1', 'class 2', 'class 4']):
+    ax = axarr[ix // 2, ix % 2]
+    # convert dataframe to long-form to plot it
+    plot_df = (
+        results_loss_df[results_loss_df.class_name == class_name]
+          .drop(columns='class_name')
+          .dropna(axis='index')
+          .reset_index()
+          .rename(columns={'index': 'identifier'})
+          .melt(id_vars='identifier', value_name='distance', var_name='num_hits')
+    )
+    sns.boxplot(data=plot_df, x='num_hits', y='distance', ax=ax,
+                order=['none/one', 'both/one', 'both/none'])
+    if preprocessing == 'pca':
+        if data_type == 'expression':
+            ax.set_ylim(0, 180)
+        elif data_type == 'rppa':
+            ax.set_ylim(0, 20)
+    elif data_type == 'rppa':
+        ax.set_ylim(0, 15)
+    elif data_type == 'expression':
+        ax.set_ylim(0, 1.75e6)
+    ax.set_title('Distributions of {} distance for {} genes, copy loss, {} data'.format(
+                   centroid_method, class_name, data_type))
+
+
+# In[18]:
+
+
+sns.set({'figure.figsize': (18, 12)})
+
+fig, axarr = plt.subplots(2, 2)
+
+# plot copy gain results here
+for ix, class_name in enumerate(['class 1', 'class 3', 'class 4']):
+    ax = axarr[ix // 2, ix % 2]
+    # convert dataframe to long-form to plot it
+    plot_df = (
+        results_gain_df[results_gain_df.class_name == class_name]
+          .drop(columns='class_name')
+          .dropna(axis='index')
+          .reset_index()
+          .rename(columns={'index': 'identifier'})
+          .melt(id_vars='identifier', value_name='distance', var_name='num_hits')
+    )
+    sns.boxplot(data=plot_df, x='num_hits', y='distance', ax=ax,
+                order=['none/one', 'both/one', 'both/none'])
+    if preprocessing == 'pca':
+        if data_type == 'expression':
+            ax.set_ylim(0, 180)
+        elif data_type == 'rppa':
+            ax.set_ylim(0, 20)
+    elif data_type == 'rppa':
+        ax.set_ylim(0, 15)
+    elif data_type == 'expression':
+        ax.set_ylim(0, 1.75e6)
+    ax.set_title('Distributions of {} distance for {} genes, copy gain, {} data'.format(
+                   centroid_method, class_name, data_type))
+
+
+# ### Individual examples
+# 
+# Even though the overall results don't really meet our expectations, are there examples of genes/cancer types that do?
```

Comment:
Maybe write more about what your expectations were? I look at those barcharts and think "okay, there's more relationship between none/one and both/one than there is between both/none, that makes intuitive sense to me."